bin/network-monitor/src/main.rs

```diff
@@ -38,306 +39,259 @@ pub const COMPONENT: &str = "miden-network-monitor";
 /// unexpectedly, the entire process exits.
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
-    // Load configuration from environment variables
-    let config = match MonitorConfig::from_env() {
-        Ok(config) => {
-            info!("Loaded configuration: {:?}", config);
-            config
-        },
-        Err(e) => {
-            anyhow::bail!("failed to load configuration: {e}");
-        },
-    };
+    // Load configuration from command-line arguments and environment variables
+    let config = MonitorConfig::parse();
+    info!("Loaded configuration: {:?}", config);
 
     if config.enable_otel {
         miden_node_utils::logging::setup_tracing(OpenTelemetry::Enabled)?;
     } else {
         miden_node_utils::logging::setup_tracing(OpenTelemetry::Disabled)?;
     }
 
-    let mut tasks = JoinSet::new();
-    let mut component_ids: HashMap<_, String> = HashMap::new();
+    let mut tasks = Tasks::new();
 
     // Initialize the RPC Status endpoint checker task.
-    let rpc_rx = initialize_rpc_status_checker(&config, &mut tasks, &mut component_ids).await;
+    let rpc_rx = tasks.spawn_rpc_checker(config.clone()).await?;
 
     // Initialize the prover checkers & tests tasks.
-    let prover_rxs = initialize_prover_tasks(&config, &mut tasks, &mut component_ids).await;
+    let prover_rxs = tasks.spawn_prover_tasks(config.clone()).await?;
 
     // Initialize the faucet testing task.
-    let faucet_rx = initialize_faucet_task(&config, &mut tasks, &mut component_ids);
+    let faucet_rx = if config.faucet_url.is_some() {
+        Some(tasks.spawn_faucet(config.clone()))
+    } else {
+        info!("Faucet URL not configured, skipping faucet testing");
```

Comment:
Would maybe make this a warn because we would probably never mean to configure it this way.